posts/my-journey-with-creating-my-first-slackbot.md

```diff
@@ -0,0 +1,199 @@
+
+During my online research, I read an article of someone that appeared to be a Slack API employee or partner that said they are aware of the issue and will update the documentation on their website. However, that didn’t help me at the time, therefore, I concluded, I wasn’t going to get anywhere by just googling the error message.
+
+
+### How I Was Able To Solve The Scheduling A Future Post Bug Issue
+
+I chose to go another route and do some research and digging into the Javascript date and time function. After my research, I came up with an idea to use an integer for the `post_at:` value, and replace the `tomorrow.getTime() / 1000` in the code snippet below.
+
+``` javascript
+// Time to post message, in Unix Epoch timestamp format
+    post_at: tomorrow.getTime() / 1000
+```
+
+Ah ha, I had a new error message now (see below); one that gave me more details and information than before. Based on the new error message `time_in_past`; that was an indicator for me that I was on the right track or in the ball park. 
+
+`Error: An API error occurred: time_in_past`
+
+Based on the above error message, I took a new path to create an integer of the exact seconds of when I wanted the post to take place. Therefore, I calculated the amount of seconds from 1970, based on the starting year of the [Javascript Date() function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date), to the date and time I wanted the message to post. Using a calculator or course, I started with 60 seconds in a minute, to the amount of seconds in a day. Then, I multipled the number days I needed times the number of seconds, started from January 1, 1970.  When I gave the `post_at:` value the new integer, there was no error message. I was confident, that I had a winning solution. The next day, my solution was verified when I seen my scheduled post. I included a unique message in the post that stated the date and time I scheduled the post, to make it easier to determine that it was the post I created the day before.
+
+```javascript
+    // example integer solution
+    post_at: 1631870268
+```
+
+## Wrap Up
+
+I’m not sure if this is the only solution or if the Slack API documentation has been updated since the writing of this blog. I’m also not sure if this is an issue when using the other programming languages or tools. All I know for sure is that I was happy to get 'a' solution working. In the meantime, you could create a JavaScript function that takes in parameters to calculate the amount of seconds from Janurary 1, 1970 to the current or future date and time. In a nut shell, that was my journey when creating my first Slackbot with the Slack API.
```

Comment:
```suggestion
Of course, I am not convinced that this is the only solution, or if the Slack API documentation has been updated since the writing of this blog. I am also not sure if this is an issue when using other programming languages or tools. All I know for sure is that I was happy to get **_a_** solution working. In the meantime, you can create a JavaScript function that takes in parameters to calculate the amount of seconds from January 1, 1970 to the current or future date and time. In a nut shell, that was my journey when creating my first Slackbot with the Slack API.
```